linux/just_files/just_install_functions.bsh

```diff
@@ -627,6 +627,53 @@ function pipenv-install()
           ' >> "${pipenv_activate}"
 }
 
+#**
+# .. function:: vsbuild-install
+#
+# Install Visual Studio build tools.
+#
+# :Arguments: * [``--installer {INSTALLER}``] - VS installer (default= ``${VS_BUILD_TOOLS_INSTALLER:-}``)
+#             * [``--download``] - Download VS installer
+#
+# Specifying the installer is can be used for offline installations:
+#
+# 1. Download https://aka.ms/vs/17/release/vs_BuildTools.exe
+# 2. ``vs_BuildTools.exe --layout C:\myLayoutDir --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 --lang en-US``
+# 3. ... Transfer C:\myLayoutDir ...
+# 4. Now you can use ``--installer C:\myLayoutDir\vs_BuildTools.exe``
+#
+# This currently requires admin privileged, until https://github.com/Data-Oriented-House/PortableBuildTools/issues/20 is resolved
+#**
+function vsbuild-install()
+{
+  if [ "${OS-}" != "Windows_NT" ]; then
+    echo "This is a windows only feature" &> 2
+    local JUST_IGNORE_EXIT_CODES=1
+    return 1
+  fi
+
+  local prefer_download=0
+  local VS_BUILD_TOOLS_INSTALLER=${VS_BUILD_TOOLS_INSTALLER-}
+
+  parse_args installer_extra_args \
+      --installer VS_BUILD_TOOLS_INSTALLER: \
+      --download prefer_download \
+      -- ${@+"${@}"}
+
+  # https://visualstudio.microsoft.com/downloads/
+  if [ ! -f "${VS_BUILD_TOOLS_INSTALLER}" ] && [ "${prefer_download}" != "0" ]; then
```

Comment:
Shouldn't `prefer_download` mean ignore `VS_BUILD_TOOLS_INSTALLER` and always download a fresh installer?  This instead appears to prefer an externally specified installed if available.

Reply:
I renamed this to `allow_download`. Downloading is still a part when you bring your own installer. The installer is only 50MB, but downloads (and checks for updates) GBs of msi/vsxi/cab/etc... files. And then installs them.